mostlyai/engine/_language/common.py

```diff
@@ -74,7 +75,11 @@ def load_base_model_and_config(
 
     # Load pretrained base model
     use_cache = not is_training  # KV cache is not needed during training
-    is_gpu_training = is_training and device.type == "cuda"
+    is_gpu_training = is_training and device.type == "cuda" and importlib.util.find_spec("bitsandbytes") is not None
+    if is_training and device.type == "cuda" and importlib.util.find_spec("bitsandbytes") is None:
+        _LOG.warning(
+            "CUDA device was found but bitsandbytes is not available. Please use extra [gpu] to install bitsandbytes for quantization."
+        )
```

Comment:
```
is_gpu_training = is_training and device.type == "cuda"
if is_gpu_training and importlib.util.find_spec("bitsandbytes") is None:
    _LOG.warning(
            "CUDA device was found but bitsandbytes is not available. Please use extra [gpu] to install bitsandbytes for quantization."
        )
        is_gpu_training = False
 ```

I find this phrasing a bit more elegant.

Reply:
sorry for the confusion, I missed we test both for `None` and `not None` on  `importlib.util.find_spec("bitsandbytes")`. I'd then anyway introduce `is_bitesandbytes_available` in order to make reading easier.